### ISSUE TITLE
Screen reader is announcing incorrect information for SwatchColorPicker

### ISSUE THREAD

**Issue by gs-ysingh** [microsoft/fluentui#21842]
### Environment Information

- **Package version(s)**: ^8.15.1
- **Browser and OS versions**: Edge Version 98.0.1108.56 (Official build) (64-bit) / Windows 11

### Describe the issue:

Screen reader is announcing incorrect information as read only when focus navigates to colors buttons. Screen reader does not announce "not selected" and role as button for color element.

### Please provide a reproduction of the issue in a codepen:

On the documentation site, the example for "Simple swatch color picker with multiple rows and larger cells that updates its icon color and shows a preview color" demonstrates the issue:

https://codepen.io/pen?&prefill_data_id=2fa24b77-379e-4380-88d8-a0db6e42c5cd

1. Load Narrator.
2. Select a color swatch inside the example with multiple rows.
3. Notice that Narrator says the option is "read only".
4. Notice that Narrator does not says "Not selected" and role "button"

#### Actual behavior:

Screen reader is announcing incorrect information as read only when focus navigates to colors buttons. The observed issues were:

1. Incorrect information such as 'Read only' is announced for interactive elements.
2. State as 'Not Selected' not announced for not selected colors.
3. Also, Role is not announced as button for the color controls.

#### Expected behavior:
Screen reader should announce correct information when focus navigates to colors buttons. The expected behavior would be:

1. Incorrect information such as Read only should not be announced for interactive elements.
2. State as 'Not Selected' should be announced for not selected colors.
3. Also, Role should be defined and announced as button for the color controls.

### Documentation describing expected behavior
https://msfast.visualstudio.com/FAST%20Experiences/_workitems/edit/450078

**Comment by gs-ysingh** [comment]
Any update on this?

**Comment by tomi-msft** [comment]
@gs-ysingh  I am having trouble exactly reproing the bug as described in this issue. Can you provide a working codepen with the `SwatchColorPicker` bug?

**Comment by smhigley** [comment]
The SwatchColorPicker with more than one row of colors is semantically a selectable grid, which is why button roles are not announced, and `button` would not be an appropriate role in any case. Narrator says "read only" for all grid cells by default. This is partly a Narrator quirk, but also related to ambiguity around what counts as "readonly" vs. editable in a grid cell -- essentially since selection isn't the same as text editing, these are treated as readonly by Narrator.

Regarding selected/not selected announcements, different screen readers will announce selection state differently (and sometimes it varies by role as well). NVDA and JAWS will generally announce "not selected", but won't announce "selected". On a grid, Narrator announces "selected", but doesn't announce "not selected". This is a Narrator UX choice -- there's a possibility it's a bug, but I wouldn't assume so.

Hope that helps! If testers are telling you something different, I'd recommend pushing back.
